### PR TITLE
CP-2108 update poynt sample app catalog

### PR DESCRIPTION
**Pull request by artemkopan**
https://poyntc.atlassian.net/browse/CP-2108

https://github.com/poynt/PoyntSamples/assets/7382939/4ae12e87-5eda-4d6a-98bf-33bf46a1ef09

https://github.com/poynt/PoyntSamples/assets/7382939/ee9082c0-1b7f-4dbb-8e2e-07710c00b252

https://github.com/poynt/PoyntSamples/assets/7382939/4a45b33d-7788-42be-84b3-d4142a8b5ecf

https://github.com/poynt/PoyntSamples/assets/7382939/4df0a20f-6661-4f53-b854-592c3ab62794

### CODE REVIEW

**Review by flongo-godaddy — APPROVED** [review]

File: commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/catalog/category/CategoryViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        CommerceDependencyProvider.getContext().subscribeOnUpdates(
+            CatalogIntents.ACTION_CATEGORIES_CHANGED
+        ).onEach {
+            loadCategories()
+        }.launchIn(viewModelScope)
+    }
+
+
+    private fun subscribeOnSearch() {
+        searchQueryFlow.debounce(300)
```
Comment (dshuba-goddady) [inline commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/catalog/category/CategoryViewModel.kt:46]: nit: I'd suggest putting value "300" to a separate constant variable such as
```kotlin
private const val SEARCH_DEBOUNCE_DELAY = 300L

...

private fun subscribeOnSearch() {
        searchQueryFlow.debounce(SEARCH_DEBOUNCE_DELAY)
        ...
}
```

File: commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/catalog/category/CategoryViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        execute {
+            val service = serviceClient.getService().getOrThrow()
+            val bundle = Bundle().apply {
+                // data source defines data provider: local db, remote or remote only if there are no data in local db.
+                // It is better to use REMOTE_IF_EMPTY in most cases to improve UX and performance.
+                putParcelable(CategoryParams.DATA_SOURCE, DataSource.REMOTE_IF_EMPTY)
+
+                // Add pagination to improve UX and avoid TooLargeTransactionException
+                putInt(CategoryParams.PAGE_OFFSET, 0)
+                putInt(CategoryParams.PAGE_SIZE, 100)
```
Comment (dshuba-goddady) [inline commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/catalog/category/CategoryViewModel.kt:62]: Same for this "[MagicNumber](https://detekt.dev/docs/rules/style#magicnumber)" value

```kotlin
private const val CATEGORY_DEFAULT_PAGE_SIZE = 100

...

    putInt(CategoryParams.PAGE_SIZE, CATEGORY_DEFAULT_PAGE_SIZE)


```

File: commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/catalog/category/create/CategoryCreateFragment.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    val selectedProduct by observableField(
+        stateFlow = { viewModel.stateFlow },
+        map = CategoryCreateViewModel.State::selectedProduct
+    )
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        bindOnCommonViewModelUpdates(viewModel)
+        launch {
```
Comment (dshuba-goddady) [inline commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/catalog/category/create/CategoryCreateFragment.kt:31]: 1. Can we combine 2 `launch { ... }` executions into one?
```kotlin
launch {
    viewModel.stateFlow.bindTo(
        map = { products to showProductDialog },
        update = ::showProductsDialog
    )

    viewModel.stateFlow.bindTo(CategoryCreateViewModel.State::createdId) { id
        id ?: return@bindTo
        Toast.makeText(
            requireContext(),
            "Category with id [$id] was created",
            Toast.LENGTH_SHORT
         ).show()
    }
}
```

2. Can we move them to a separate function?

File: commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/catalog/priceAdjustment/PriceAdjustmentFragment.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+class PriceAdjustmentFragment :
+    CommonFragment<PriceAdjustmentFragmentBinding>(R.layout.price_adjustment_fragment) {
+
+
+    private val viewModel: PriceAdjustmentViewModel by viewModels()
+
+    val items by observableField(
+        stateFlow = { viewModel.stateFlow },
+        map = { items.map { it.mapToUiItems() } },
+        keySelector = { it.items })
```
Comment (dshuba-goddady) [inline commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/catalog/priceAdjustment/PriceAdjustmentFragment.kt:21]: nit: right parenthesis `)` should be at list in the next line
```kotlin
val items by observableField(
    stateFlow = { viewModel.stateFlow },
    map = { items.map { it.mapToUiItems() } },
    keySelector = { it.items }
)
```

File: commerce-services-sample/src/main/res/layout/category_create_fragment.xml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                android:layout_marginTop="4dp"
+                android:onTextChanged="@{(s, start, before, count) -> viewModel.onDescriptionChanged(s.toString())}" />
+
+
+            <TextView
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginHorizontal="@dimen/item_spacing"
+                android:layout_marginTop="8dp"
+                android:text="@{@string/category_product_added + ` ` + fragment.selectedProduct.name}"
```
Comment (dshuba-goddady) [inline commerce-services-sample/src/main/res/layout/category_create_fragment.xml:70]: nit: Would it be possible to use formatted string instead? So the value is placed into %s placeholder in the string resource

**Review by dshuba-goddady — APPROVED** [review]
Left a few minor optional nit comments, but overall everything looks good

File: commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/catalog/category/create/CategoryCreateFragment.kt (hunk 3 above)
Comment (artemkopan) [reply commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/catalog/category/create/CategoryCreateFragment.kt:31]: 1.. we can't combine it into one. if we combine then viewModel.stateFlow.bindTo(CategoryCreateViewModel.State::createdId) would never called.

2.done

File: commerce-services-sample/src/main/res/layout/category_create_fragment.xml (hunk 5 above)
Comment (artemkopan) [reply commerce-services-sample/src/main/res/layout/category_create_fragment.xml:70]: It is a sample app, should we care about formatting?

**Review by ylutsenko-godaddy — APPROVED** [review]

**Review by mvideshyn-godaddy — APPROVED** [review]

**Review by dshuba-goddady — APPROVED** [review]

File: commerce-services-sample/src/main/res/layout/category_create_fragment.xml (hunk 5 above)
Comment (dshuba-goddady) [reply commerce-services-sample/src/main/res/layout/category_create_fragment.xml:70]: As a "nit", we can consider it optional or preferable so it's okay to also keep it as it is

Comment (lalkhawaja-godaddy) [reply commerce-services-sample/src/main/res/layout/category_create_fragment.xml:70]: Why not :)  be a good model for 3rd party app as you for our team @artemkopan

**Pull request merged**